### ISSUE TITLE
GPT to provide answers only from the provided sources?

### ISSUE THREAD

**Issue by viru2206** [Azure-Samples/azure-search-openai-demo#564]
Is there any way we can get answer only from the provided sources. I tried giving a proper command in the system message of chatreadretrieveread script but it still provides me with the answers and it provides citations also. Yeah it happens less now but its still happening sometimes.

**Comment by GauravSingh1A** [comment]
Yeah, I would also love to know how we can do that.

**Comment by pamelafox** [comment]
Can you share examples of where it's answering questions that aren't in the provided sources? Have you tried using GPT-4? (It is generally better at sticking to instructions).

**Comment by viru2206** [comment]
Yeah sure, I have attached an image in which you can see that it provided reponse that are not from PDF and also gave citations. The citation does not have any information regarding the query. Also, the GPT knows that the answers are not from the sources. No, I haven't tried GPT-4, I have applied for GPT-4 access but hasn't got yet.
![Screenshot (314)](https://github.com/Azure-Samples/azure-search-openai-demo/assets/92709298/55e8fbe3-8d18-4ff3-bf86-4d5ba0f7660f)

**Comment by shamprasadrh** [comment]
can you try setting temperature to 0 ?

**Comment by viru2206** [comment]
Yeah, I already tried that.

**Comment by Kesavaraja** [comment]
I am also facing the same issue. I got the content and citations from Wikipedia. Which is weird for me.

**Comment by Kesavaraja** [comment]
I am also facing the same issue. I get the content and citations from Wikipedia. Which is weird for me.

**Comment by FranciscoMMendes** [comment]
I am also facing the same issue on my side. I have tried to improve the prompt but it does not change the reply and the citations provided.
When I ask the chat a question about some personality or celebrity for example, the chat fetches the information from Wikipedia instead of searching the documents provided.

**Comment by chriskabat63** [comment]
I added a little more to the question prompt and seem to get pretty good results.  It seems to still honor the system message, but I think this helps reinforce that it should use the included sources.  

Here is kind of what the prompt looks like now:

```
Answer the following question only using the sources listed below:

Question: 1933 superbowl winner

Sources:

Source 1: ….

Source 2: …
```


I basically just replaced line 179:
https://gist.github.com/chriskabat63/59da4f18657ab2b2f043cedd18d1c9ed

**Comment by pamelafox** [comment]
Are you all still seeing this? We may move to a slightly tweaked prompt that emphasizes coherence to sources more, you could try this prompt:

https://github.com/Azure-Samples/azure-search-openai-demo/pull/967/files#diff-d6c03003d61bd0a593ea552fb76573df32d24fe6ac495da5ac2f7250cc372b96

**Comment by pamelafox** [comment]
Thanks for raising this. Since this was filed, the app’s prompt was tightened (see chat_answer_question.prompty) to answer only from retrieved sources and say “I don’t know” otherwise, and we added evaluation tools for groundedness and citation matching. 

It's still always the case that LLMs can generate un-grounded answers, due to the way they're designed. However, I recently ran evaluations using the new GPT-5 models, and I found *no* answering-outside-sources with the full GPT-5 model. (I saw a few with gpt-5 mini). See my eval write-up here:
https://techcommunity.microsoft.com/blog/azuredevcommunityblog/gpt-5-will-it-rag/4442676

Those evals are also checked into the repo, in the evals folder. I recommend trying gpt-5 to minimize the chance of answering outside sources.
